controllers/osbuildenvconfig_controller.go

```diff
@@ -490,20 +660,323 @@ func (r *OSBuildEnvConfigReconciler) generateComposerService(serviceName, portNa
+	workerConfigInventoryParams := struct {
+		Address string
+		User    string
+		SSHKey  string
+	}{
+		Address: workerAddress,
+		User:    workerUser,
+		SSHKey:  path.Join(workerConfigJobSSHKeyDir, workerSSHKeysPrivateKeyKey),
+	}
+	return r.ensureConfigMapForTemplateFileExists(ctx, fmt.Sprintf(workerConfigInventoryConfigMapNameFormat, workerName), workerConfigInventoryConfigMapKey, workerConfigInventoryTemplateFile, workerConfigInventoryParams, instance)
+}
+
+func (r *OSBuildEnvConfigReconciler) ensureOSBuildWorkerConfigExists(ctx context.Context, instance *osbuildv1alpha1.OSBuildEnvConfig) (bool, error) {
+	workerOSBuildWorkerConfigParams := struct {
+		CredentialsFile string
+		Bucket          string
+		GenericS3       *struct {
+			Region              string
+			Endpoint            string
+			CABundleFile        *string
+			SkipSSLVerification *bool
+		}
+	}{
+		CredentialsFile: workerOSBuildWorkerConfigS3CredentialsFile,
+	}
+
+	if instance.Spec.S3Service.AWS != nil {
+		workerOSBuildWorkerConfigParams.Bucket = instance.Spec.S3Service.AWS.Bucket
+	} else {
+		workerOSBuildWorkerConfigParams.Bucket = instance.Spec.S3Service.GenericS3.Bucket
+		workerOSBuildWorkerConfigParams.GenericS3 = &struct {
+			Region              string
+			Endpoint            string
+			CABundleFile        *string
+			SkipSSLVerification *bool
+		}{
+			Region:   instance.Spec.S3Service.GenericS3.Region,
+			Endpoint: instance.Spec.S3Service.GenericS3.Endpoint,
+		}
+		if instance.Spec.S3Service.GenericS3.CABundleSecretReference != nil {
+			caBundleFile := workerOSBuildWorkerConfigS3CABundleFile
+			workerOSBuildWorkerConfigParams.GenericS3.CABundleFile = &caBundleFile
+		}
+		if instance.Spec.S3Service.GenericS3.SkipSSLVerification != nil {
+			workerOSBuildWorkerConfigParams.GenericS3.SkipSSLVerification = instance.Spec.S3Service.GenericS3.SkipSSLVerification
+		}
+	}
+	return r.ensureConfigMapForTemplateFileExists(ctx, workerOSBuildWorkerConfigConfigMapName, workerOSBuildWorkerConfigConfigMapKey, workerOSBuildWorkerConfigTemplateFile, workerOSBuildWorkerConfigParams, instance)
+}
+
+func (r *OSBuildEnvConfigReconciler) ensureWorkerSetupJobExists(ctx context.Context, instance *osbuildv1alpha1.OSBuildEnvConfig, workerName, workerSSHKeySecretName string) (bool, error) {
+	_, err := r.JobRepository.Read(ctx, fmt.Sprintf(workerConfigJobNameFormat, workerName), conf.GlobalConf.WorkingNamespace)
+	if err == nil {
+		return false, nil
+	}
+
+	if errors.IsNotFound(err) {
+		workerSetupJob, err := r.generateWorkerSetupJob(workerName, workerSSHKeySecretName, instance)
+		if err != nil {
+			return false, err
+		}
+
+		err = r.JobRepository.Create(ctx, workerSetupJob)
+		if err != nil {
+			return false, err
+		}
+
+		return true, nil
+	}
+
+	return false, err
+}
+
+func (r *OSBuildEnvConfigReconciler) generateWorkerSetupJob(workerName, workerSSHKeySecretName string, instance *osbuildv1alpha1.OSBuildEnvConfig) (*batchv1.Job, error) {
+	workerSetupJobParams := struct {
```

Comment:
again - I think it will be better to export is as a type

Reply:
I don't really see the benefit, but I made the change